etp-backend/src/test/clj/solita/etp/service/laatija_test.clj

```diff
@@ -9,19 +9,40 @@
 
 (t/use-fixtures :each ts/fixture)
 
-(def laatija-generators {common-schema/Henkilotunnus (g/always "130200A892S")
+(defn unique-henkilotunnus-range [to]
+  (->> (range 0 to)
+       (map (partial format "%09d"))
+       (map #(str % (common-schema/henkilotunnus-checksum %)))
+       (map #(str (subs % 0 6) "-" (subs % 6 10)))))
+
+(def laatija-generators {common-schema/Henkilotunnus      (g/always "130200A892S")
                          laatija-schema/MuutToimintaalueet
-                         (g/always [0,1,2,3,17])
-                         common-schema/Date (g/always (java.time.LocalDate/now))})
+                                                          (g/always [0, 1, 2, 3, 17])
+                         common-schema/Date               (g/always (java.time.LocalDate/now))})
 
 (t/deftest add-and-find-test
-  (doseq [laatija (repeatedly 100 #(g/generate schema/LaatijaSave laatija-generators))
-          :let [id (service/add-laatija! ts/*db* laatija)]]
-    (t/is (= (assoc laatija :id id) (service/find-laatija ts/*db* id)))))
+  (let [laatija-count    100
+        henkilotunnukset (unique-henkilotunnus-range laatija-count)
+        laatijat         (mapv (fn [laatija henkilotunnus]
+                                 (assoc laatija :henkilotunnus henkilotunnus)) (repeatedly laatija-count #(g/generate schema/LaatijaSave laatija-generators)) henkilotunnukset)
+        added-laatijat   (service/add-or-update-existing-laatijat! ts/*db* laatijat)]
+    (doseq [[idx laatija-id] (map-indexed vector added-laatijat)]
+      (t/is (= (assoc (get laatijat idx) :id laatija-id) (service/find-laatija ts/*db* laatija-id))))))
+
+(t/deftest add-and-update-existing-test
```

Comment:
Good usage of schemas here! Maybe it would be possible to add another laatija to database to test that only one laatija got updated, not all of them.